File: vigil-backend/app/schemas/analysis.py
from datetime import datetime
from typing import List, Optional
import uuid
from pydantic import BaseModel, ConfigDict
from app.schemas.base import PaginatedResponse


class AnalysisRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    pull_request_id: uuid.UUID
    head_sha: str
    status: str
    trigger_type: str
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    error_message: Optional[str] = None
    created_at: datetime


class AnalysisListResponse(PaginatedResponse):
    items: List[AnalysisRead]

File: vigil-frontend/src/types/analysis.ts
import type { PaginatedResponse } from './api';

export type AnalysisStatus = 'QUEUED' | 'RUNNING' | 'COMPLETED' | 'FAILED' | 'CANCELLED';
export type AnalysisTrigger = 'WEBHOOK' | 'MANUAL';

export interface AnalysisRead {
  id: string; // uuid
  pull_request_id: string; // uuid
  head_sha: string;
  status: AnalysisStatus | string;
  trigger_type: AnalysisTrigger | string;
  started_at: string | null;
  completed_at: string | null;
  error_message: string | null;
  created_at: string;
}

export type AnalysisListResponse = PaginatedResponse<AnalysisRead>;
